transport.go

```diff
@@ -777,13 +777,17 @@ type async chan interface{}
 func (p async) await(ctx context.Context) (Response, error) {
 	select {
 	case x := <-p:
-		switch v := x.(type) {
-		case Response:
-			return v, nil
-		case error:
-			return nil, v
-		default:
-			panic(fmt.Errorf("BUG: promise resolved with impossible value of type %T", v))
+		if x == nil {
```

Comment:
Would it be possible to put `case nil` in the `switch` below rather than have a separate block here?

Reply:
np